### PR TITLE
feat: add a map for retrieving the component when rendering a cell type

### PR DESCRIPTION
**Pull request by cindyli**

### CODE REVIEW

File: docs/DeveloperDoc.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Developer Documentation
+
+## How to render a palette
+
+`Palette.ts` accepts a JSON in a structure demostrated in
```
Comment (klown) [inline docs/DeveloperDoc.md]: Spelling error "demostrated".

Here is a suggested rewording of the first paragraph up to the third sentence:

"
`Palette.js` constructs a palette based on a JSON file that contains a list of all of the cells in the palette.  An example is found in [`src/keyboards/bmw_palette.json`](../src/keyboards/bmw_palette.json). The `cells` object is the list of all of the cells.  Each cell has a `type` key and an `options` key.
"

The rest of the paragraph is fine.

File: docs/DeveloperDoc.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+[`src/keyboards/bmw_palette.json`](../src/keyboards/bmw_palette.json). In the
+`cells` object, each cell has a `type` key and a `options` key. The `type`
+value indicates which component should be used to render this cell. The
+`options` contains information to be passed to the component.
+
+When a new `type` value is introduced, a developer need to:
```
Comment (klown) [inline docs/DeveloperDoc.md]: There is a mismatch between singular subject, "a developer", and plural verb, "need to".  I suggest using plural throughout:

"
When a new `type` value is introduced, developers need to:
"

**Review by klown — CHANGES_REQUESTED** [review]
Thanks for taking this registry off of my task list, @cindyli !

The code changes are fine -- thanks also for fixing my renaming goof :-(

I've made suggested rewording for the developer documentation.  You might have a better way to reword.  I'll leave that to you.

**Comment by cindyli** [comment]
Thanks for the review, @klown. Your wording is much better. All comments are addressed. Ready for another look.

File: docs/DeveloperDoc.md [hunk 3]
```diff
@@ -2,13 +2,15 @@
 
 ## How to render a palette
 
-`Palette.ts` accepts a JSON in a structure demostrated in
-[`src/keyboards/bmw_palette.json`](../src/keyboards/bmw_palette.json). In the
-`cells` object, each cell has a `type` key and a `options` key. The `type`
-value indicates which component should be used to render this cell. The
-`options` contains information to be passed to the component.
+`Palette.ts` constructs a palette based on a JSON file that contains a list of
+of the cells in the palette. An example is found in
```
Comment (klown) [inline docs/DeveloperDoc.md:6]: Two "of"s here.  One is at the end of the previous line just before the carriage return, and the other is just after at the beginning of the next line.

Comment (cindyli) [reply docs/DeveloperDoc.md:6]: Good catch. Fixed now. Thanks.

**Review by klown — APPROVED** [review]
LGTM.  I'll merge it with my branch

**Pull request merged**